pkg/dataflatten/json.go

```diff
@@ -4,13 +4,26 @@ import (
 	"encoding/json"
 	"fmt"
 	"os"
+
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
 )
 
 func JsonFile(file string, opts ...FlattenOpts) ([]Row, error) {
 	rawdata, err := os.ReadFile(file)
 	if err != nil {
 		return nil, err
 	}
+
+	// Attempt to decode utf16 data.
+	valid_json := json.Valid(rawdata)
+	if !valid_json {
+		rawdata, _, err = transform.Bytes(unicode.UTF16(unicode.LittleEndian, unicode.UseBOM).NewDecoder(), rawdata)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return Json(rawdata, opts...)
 }
```

Comment:
Code is fine, but I have a style nit. We typically try to reduce nesting as much as possible, so you could do something like this:

```suggestion
        if json.Valid(rawdata) {
            return Json(rawdata, opts...)
        }
        
        // invalid json
        rawdata, _, err = transform.Bytes(unicode.UTF16(unicode.LittleEndian, unicode.UseBOM).NewDecoder(), rawdata)
        if err != nil {
	    return nil, err
	}

	return Json(rawdata, opts...)
}
```

Reply:
this is also a good suggestion